### ISSUE TITLE
posibility of a 5.0.3 release with less strict node-gyp version dependency 

### ISSUE THREAD

**Issue by paimon0715** [mapbox/node-sqlite3#1493]
Hi, @kewdej, there are two high severity vulnerabilities introduced by **tar@2.2.2**:
### Issue Description

   I noticed that a vulnerability is introduced in **_sqlite3@5.0.2_**:
     Vulnerability **CVE-2021-32804** and **CVE-2021-32803** (high severity) affects package **_tar_** (versions:<3.2.2,>=4.0.0 <4.4.14,>=5.0.0 <5.0.6,>=6.0.0 <6.1.1):  [https://snyk.io/vuln/SNYK-JS-TAR-1536531](https://snyk.io/vuln/SNYK-JS-TAR-1536531) and [https://snyk.io/vuln/SNYK-JS-TAR-1536528](https://snyk.io/vuln/SNYK-JS-TAR-1536528)
  The above vulnerable package is referenced by **_sqlite3@5.0.2_** via:
`sqlite3@5.0.2 ➔ node-gyp@3.8.0 ➔ tar@2.2.2`

Since **_sqlite3@5.0.2_** ([214,273 downloads per week](https://www.npmjs.com/package/sqlite3/v/5.0.2?activeTab=versions)) is referenced by **8,775** downstream projects (e.g., [websql 2.0.2](https://www.npmjs.com/package/websql) (latest version), [@sap/cds-dk 4.4.1](https://www.npmjs.com/package/@sap/cds-dk) (latest version), [typeorm-model-generator 0.4.5](https://www.npmjs.com/package/typeorm-model-generator) (latest version), [ueberdb2 1.4.13](https://www.npmjs.com/package/ueberdb2) (latest version), [indexeddbshim 8.0.0](https://www.npmjs.com/package/indexeddbshim) (latest version)), the above vulnerabilities  can be propagated into these downstream projects and expose security threats to them via the following package dependency paths:
 (1)`@grouparoo/core@0.5.2 ➔ sqlite3@5.0.2 ➔ node-gyp@3.8.0 ➔ tar@2.2.2`
(2)`@contrast/test-bench-utils@3.20.1-alpha.0 ➔ sqlite3@5.0.2 ➔ node-gyp@3.8.0 ➔ tar@2.2.2`
  **......**

If **_sqlite3@5.0.\*_** removes the vulnerable package from the above version, then its fixed version can help downstream users decrease their pain.

**Given the large number of downstream users, could you help update your package to remove the vulnerability from sqlite3@5.0.2 ?**

### Fixing suggestions

In **_sqlite3@5.0.3_**, maybe you can kindly try to perform the following upgrade :
`node-gyp 3.x ➔ ^4.0.0`; 

**Note:**
_**node-gyp@4.0.0**(>=4.0.0) directly depends on **tar@4.4.19** which has fixed the vulnerability [**CVE-2021-32804**](https://snyk.io/vuln/SNYK-JS-TAR-1536531) and [**CVE-2021-32803**](https://snyk.io/vuln/SNYK-JS-TAR-1536528)._

Thank you for your attention to this issue and welcome to share other ways to resolve the issue.

Best regards,
^_^

**Comment by Trinovantes** [comment]
There doesn't seem to be any breaking changes in `node-gyp` from `3.x` to its latest version `^8` (at least for `node-sqlite3` specifically). I resolved the security warnings by explicitly adding this to my package.json

```json
"resolutions": {
  "node-gyp": "^8"
}
```

**Comment by unuseless** [comment]
Looking forward to a fix for the tar dependency.

**Comment by 564064202** [comment]
node-gyp the version is too low! this effects other packages,it like a cancer

**Comment by sharedrory** [comment]
Please update `node-gyp` version, security should not be taken lightly.

**Comment by teklakct** [comment]
@564064202 @sharedrory it is updated already but only on master/main branch. There is no new tagged version released yet. 
You can specify a "fix" commit if you want. Simply add this to your dependencies in `package.json`

`"sqlite3": "git@github.com:mapbox/node-sqlite3.git#3fb3715c4e699ca3bc19e47b136758c66f61e477",`

**Comment by sharedrory** [comment]
@teklakct Thank you. I looked at the commit history and since the current release there have been some bug fixes and the (pre)node-gyp update. While your suggestion works, I think something like a vulnerability fix requires its own emergency release.

**Comment by teklakct** [comment]
Yes, it should but I cannot wait so we decide to specify commit instead

**Comment by sharedrory** [comment]
Oh sorry I wasn't clear. Yes people that want to have it fixed asap should use that method (or fork) but I meant its the developers/maintainers responsibility to keep the dependencies up-to-date.

**Comment by 564064202** [comment]
please publish to npm

**Comment by paridhi-agr** [comment]
Hi @kewde, would request you to please release this fix in 5.0.2 revision. It is causing high severity vulnerabilities. Thanks!

**Comment by richardj-bsquare** [comment]
Few more issues since this was first opened....
``` bash
$ yarn add sqlite3 
warning sqlite3 > node-pre-gyp@0.11.0: Please upgrade to @mapbox/node-pre-gyp: the non-scoped node-pre-gyp package is deprecated and only the @mapbox scoped package will recieve updates in the future
warning sqlite3 > node-gyp > request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
warning sqlite3 > node-gyp > request > har-validator@5.1.5: this library is no longer supported
```

**Comment by rufreakde** [comment]
Will the fix be published soon?

**Comment by rickbergfalk** [comment]
Haven't tried it yet, but it seems vscode folks published a fork with security fixes https://www.npmjs.com/package/@vscode/sqlite3

**Comment by davedoesdev** [comment]
@rickbergfalk thanks for the heads up, it works for me. I also learned about npm aliases which made it easier to switch:

```json
    "sqlite3": "npm:@vscode/sqlite3@^5.0.7",
```

**Comment by josep11** [comment]
> @rickbergfalk thanks for the heads up, it works for me. I also learned about npm aliases which made it easier to switch:
> 
> ```json
>     "sqlite3": "npm:@vscode/sqlite3@^5.0.7",
> ```

Tried this one and works perfectly fine for me.
But just for you to beware it depends on Python >= 3.6.0. I noticed it on CircleCI builds.

**Comment by kirillgroshkov** [comment]
please publish master to npm! 🙏

**Comment by ThePeterMick** [comment]
Any updates when this pkg will be update?

Apparently the fix was in https://github.com/mapbox/node-sqlite3/commit/3fb3715c4e699ca3bc19e47b136758c66f61e477

**Comment by the1corndog** [comment]
Hello,

Is there a reason why this vulnerability hasn't been released yet? Seems the problem was resolved in the code base?

Thank you

**Comment by daniellockyer** [comment]
`sqlite3` v5.0.3 has been shipped 🙂

**Comment by kberg** [comment]
Thanks @daniellockyer!
